lib/uplinkc/object.go

```diff
@@ -148,7 +148,9 @@ func upload_write(uploader C.UploaderRef, bytes *C.uint8_t, length C.size_t, cEr
 
 	n, err := upload.wc.Write(buf)
 	if err != nil {
-		*cErr = C.CString(fmt.Sprintf("%+v", err))
```

Comment:
We don't want to set the error if it's not cancelled here? There was no check before.

Reply:
We don't set the error if it **is** canceled. These changes make `upload_cancel`, `upload_commit`, `upload_write`, `download_close`, `download_cancel`, and `download_read` all behave the same way in that they return early if they encounter errors but they only set the error string if that error is not a context cancellation.

I'm not certain that this is the best approach but the inconsistency was causing issues. If we keep it this way I think it might make sense to be able to know from C whether your uploader/downloader is closed/comitted/canceled. I think you and I have actually talked about that before.